modules/contracts/adapters/dex/src/adapter.rs

```diff
@@ -212,7 +212,7 @@ pub trait DexAdapter: AbstractNameService + Execution {
         let pair_address = exchange.pair_address(
             deps,
             ans.host(),
-            (paired_assets.swap_remove(0), paired_assets.swap_remove(1)),
+            (paired_assets.swap_remove(0), offer_asset.name.clone()),
```

Comment:
It is unclear if this method expects the user to provide the offer asset with the paired assets or not.